spec/integration/users_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'rails_helper'
+
+RSpec.describe 'users/index', type: :feature do
+  before(:each) do
+    @user = [
+      User.create(
+        name: 'Mustafa',
+        photo: 'https://pic.com',
+        bio: 'Software Engineer Turkey',
+        postscounter: 2
+      ),
+      User.create(
+        name: 'Mert',
+        photo: 'https://pic.com',
+        bio: 'Software Engineer from Turkey',
+        postscounter: 3
+      ),
+      User.create(
+        name: 'Caney',
+        photo: 'https://pic.com',
+        bio: 'Software Engineer from England',
+        postscounter: 3
+      )
+    ]
+    visit users_path
+  end
+
+  it 'shows the author image' do
+    expect(page).to have_css('img')
+  end
+
+  it 'shows the username of all users' do
+    @user.each do |user|
+      expect(page).to have_content(user.name)
+    end
+  end
+
+  it 'shows the number of posts by each author' do
+    @user.each do |user|
+      expect(page).to have_content("Number of posts: #{user.postscounter}")
+    end
+  end
+
+  it 'redirects to users show page when clicked' do
```

Comment:
- [ ] according to the requirements there should be a Spec file that only tests User index page, a Spec file that only tests User show page, a Spec file that only tests User post index page and a Spec file that only tests User post show page.